deps/picotls/lib/openssl.c

```diff
@@ -693,77 +695,138 @@ int ptls_openssl_create_key_exchange(ptls_key_exchange_context_t **ctx, EVP_PKEY
     }
 }
 
+struct sign_ctx {
+    const struct st_ptls_openssl_signature_scheme_t *scheme;
+    EVP_MD_CTX *ctx;
+    ptls_buffer_t buf;
+    size_t siglen;
+    ASYNC_WAIT_CTX *waitctx;
+    ASYNC_JOB *job;
```

Comment:
I see this property being set but never used. Is it something expected to be used when we implement cancellation (i.e., when `ptls_free` is called while signing request in inflight)?

Reply:
The job pointer is kept because when this function is called the second time, the job is resumed via this ptr.

> To restart a job call ASYNC_start_job() again passing the job handle in *job. The func, args and size parameters will be ignored when restarting a job. https://www.openssl.org/docs/man1.1.1/man3/ASYNC_start_job.html

[Here](https://github.com/sharksforarms/h2o/blob/sharksforarms/async/lib/common/socket.c#L1460-L1465) is where I handle cancellation in h2o, maybe this is better placed in picotls? I'll give it a try.

Reply:
Re cancellation, yes I would appreciate that be done in picotls.

FWIW, `verify_certificate` callback has a mechanism for that; see https://github.com/h2o/picotls/commit/7970614ad049d194fe1691bdf0cc66c6930a3a2f/. Preferably I think we want something with similar interface.

Reply:
addressed in https://github.com/h2o/picotls/pull/408